src/model/consent/consent.ts

```diff
@@ -0,0 +1,86 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import Knex from 'knex'
+
+/*
+ * Interface for Consent resource type
+ */
+export interface Consent {
+  id: string;
+  initiatorId: string;
+  participantId: string;
+  credentialId?: number;
+  credentialType?: string;
+  credentialStatus?: string;
+  credentialPayload?: string;
+  credentialChallenge?: string;
+  timeStamp?: string;
+}
+
+/*
+ * Class to abstract Consent DB operations
+ */
+export class ConsentDB {
+  // Knex instance
+  private Db: Knex
+
+  public constructor (dbInstance: Knex) {
+    this.Db = dbInstance
+  }
+
+  // Add initial Consent parameters
+  public register (consent: Consent): Promise<Consent[]> {
+    return this.Db<Consent>('Consent')
+      .insert(consent)
+  }
+
+  // Update Consent credential
+  public updateCredentials (consent: Consent): Promise<Consent[]> {
+    return this.Db<Consent>('Consent')
+      .where({ id: consent.id })
+      .update({
+        credentialId: consent.credentialId,
```

Comment:
So my only concern here is that if you call `updateCredentials` with null or undefined fields, you could end up overwriting the existing fields.

Perhaps a better approach could be something like:
```js
const fieldsToUpdate = [ 'credentialId', 'credentialType', 'credentialStatus'...];
const updated = {}
Object.keys(consent).forEach(key => {
  if ( fieldsToUpdate.indexOf(key) > -1) {
    return;
  }
  const value = consents[key]
  if (!value) {
    return;
  }
  updated[key] = value
});

...then
.update(updated)
```

Reply:
Implemented in a recent commit.